cli/azd/cmd/up.go

```diff
@@ -99,17 +96,29 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		return nil, fmt.Errorf("running init: %w", err)
 	}
 
+	infraCreateAction, err := u.infraCreateActionInitializer()
+	if err != nil {
+		return nil, err
+	}
+
+	infraCreateAction.flags = &u.flags.infraCreateFlags
 	provisionOptions := &middleware.Options{Name: "infracreate", Aliases: []string{"provision"}}
-	_, err = u.runner.RunChildAction(ctx, provisionOptions, u.infraCreate)
+	_, err = u.runner.RunChildAction(ctx, provisionOptions, infraCreateAction)
 	if err != nil {
 		return nil, err
 	}
 
 	// Print an additional newline to separate provision from deploy
 	u.console.Message(ctx, "")
 
+	deployAction, err := u.deployActionInitializer()
+	if err != nil {
+		return nil, err
```

Comment:
Is this moving us to the place where the user would see an error on the middle of running `azd up` (like right after provision) ?
For example, if during runtime we failed initializing the action?  

Should we move all the calls to init the actions to be all together at the top, so we would fail fast?  But then, we would be losing the laziness of waiting to init.

Reply:
@vhvb1989 I think this is the right behavior. Composable actions should behave as-if user ran `init`, then run `provision` and finally `deploy`. It is not quite possible to prevalidate all preconditions of `provision` and `deploy` in the `init` phase. For example, `deploy` might require a set of tools (language frameworks). Precomputing all this complicated and might not be possible in all instances.

Reply:
@vhvb1989 - the only reason these initializers would fail is if there was a DI container registration issue which we have unit tests around that verify this.

Reply:
> IIRC, the fail fast is an important feature for azd. We want to promote azd up as a golden path, and reduce the chances for failing in the middle of the process as much as possible.

@vhvb1989 What I'm getting at is that you **do not** guarantee fail fast behavior by moving initializers later. Think about any runtime validation logic that wouldn't be captured in dependency initialization.

> Composable actions should behave as-if user ran init, then run provision and finally deploy
> We should never change or based a product requirement upon a component design limitation.

I didn't mean to state it as a product requirement. It's a user mental model about how the command should works. It is confusing to the user when user encounters "error: dotnet isn't installed" at the start, when they first expect to init the repository. That's what I mean by user should encounters error as-if they ran commands individually. It avoids exposing errors that are outside of what's expected by the user at a given phase.